### ISSUE TITLE
v0.43.05 memory layouts

### ISSUE THREAD

**Issue by wookietreiber** [splintermind/Dwarf-Therapist#311]
I'm waiting for the new memory layout. I created this issue, so we (everyone subscribing to this issue) will get notified once the new memory layout has been added and the issue gets closed.

- [ ] Linux
- [ ] OS X
- [x] Windows

(Edit: I hope this is OK this way, I can see no other means of notification other than frequently checking the repository manually.)

**Comment by n0manarmy** [comment]
I agree with this post. I'm waiting for 43.04 as well. I didn't see any local info on how to find the values too. I could dig through the forums if there is no tutorial anywhere else

**Comment by lethosor** [comment]
0.43.04 is weird because of the new Windows compiler. It's likely that there will need to be more changes than usual to make it possible to find layouts, and maybe to use DT. We (DFHack people) haven't even gotten to the point where we can auto-generate DT layouts from df-structures for 0.43.04 yet.

Linux and OS X should be about as easy as before, once we get df-structures updated - it's the next release that'll be the challenge there.

**Comment by wookietreiber** [comment]
Updated to v0.43.05.

**Comment by fchaix** [comment]
Waiting for this layout… :/

**Comment by lethosor** [comment]
It's going to take a while.

**Comment by Hello71** [comment]
make that two, no, _five_ whiles.

**Comment by hovissimo** [comment]
Anyone else who shows up and would like these layouts, you don't need to respond to the thread to get notifications, you can subscribe directly or use the thumbs up +1 reaction on the original post to show your support.

Thanks for helping to keep this thread tidy!

**Comment by Richard-Degenne** [comment]
Do we know why these layouts are so long to come out? Not being able to use Therapist is a real pain in the *ss. :'(

**Comment by Sprunth** [comment]
I'm not part of the group working on it, but almost undoubtedly a new compiler and 64-bit addresses will have made things complicated.  The difficulty will be similar to understanding the memory of a completely new program, except for the fact that we know (for the most part) what we are looking for.

**Comment by kitsuneninetails** [comment]
Looks like the layouts are still waiting?

**Comment by Lagg** [comment]
Can anyone give a list of files aside from `memorylayout.*` where the constants and code for offsets and heap reading live?

**Comment by lethosor** [comment]
Maybe `dfinstance*` and `gamedatareader.*`?

**Comment by levireeves** [comment]
http://www.bay12forums.com/smf/index.php?topic=122968.msg7222256#msg7222256

**Comment by wookietreiber** [comment]
**Note:** I edited my original post to include a task list for the supported operating systems. Since the memory for Windows has been merged, I already checked that task.

**Comment by hroest** [comment]
are there any plans to also create memory layouts for linux and OSX? Is there a description of the process on how to obtain these memory addresses somewhere, what do people usually do on Linux to obtain these layout files (e.g. does one have to use fmem ?)

**Comment by maxgraser** [comment]
Will there also be 64-bit memory layouts in the future?

**Comment by lethosor** [comment]
I use `devel/export-dt-ini` in DFHack, usually. It seems to generate 64-bit layouts just fine, although to the best of my knowledge, they haven't been tested much. I did fix one issue that was reported with what it generated on 64-bit Linux.

@hroest If you're looking for one, [here's one I generated](https://gist.github.com/lethosor/4ce2a574a510ddaadcbdfd829eac20c9). Note that it will not work with DT 37.

**Comment by bensleveritt** [comment]
@lethosor I'm not normally one for bumping for the sake of it but given that 

> Linux and OS X should be about as easy as before

is there any chance of a DT for other OSes?

**Comment by lethosor** [comment]
That was referring to 0.43.04, which was 32-bit only. Whatever existing DT builds are around for pre-0.43.04 should work for 0.43.04 on all OSes, given a valid memory layout.

**Comment by bensleveritt** [comment]
Ah, thanks. I'm guessing 64bit is much harder to crack, and it'll still be some time before we see DT for *nix OSes 0.43.05+?

**Comment by lethosor** [comment]
The hard part is getting DT to work with any 64-bit DF builds. 64-bit layouts are easy to generate, but verifying that they work can really only be done with a DT that supports 64-bit DF.

**Comment by bensleveritt** [comment]
I'm confused. Essentially. What's the latest version of DF/DT I can use on MacOS?

**Comment by lethosor** [comment]
Right now, 0.43.03 with DT 37 (or *possibly* older versions, although I can't guarantee that they would work). You would probably need to add [this layout](https://github.com/splintermind/Dwarf-Therapist/blob/DF2016/share/memory_layouts/osx/v0.43.03_osx.ini), following the instructions [here](https://github.com/splintermind/Dwarf-Therapist/tree/DF2016/share/memory_layouts). In theory, 0.43.04 would also work with DT 37, but I don't know if anyone ever bothered generating layouts for it.

**Comment by bensleveritt** [comment]
Brilliant. Thanks @lethosor

**Comment by Richard-Degenne** [comment]
Any update? It's been a while.

**Comment by int-ua** [comment]
https://github.com/Hello71/Dwarf-Therapist/blob/f25ef0fc857fc188ee9f1ec9085b6c1c751426f4/share/memory_layouts/linux/v0.43.05.ini

**Comment by lethosor** [comment]
Keep in mind that that is just for Linux, and you'll have to use https://github.com/Hello71/Dwarf-Therapist instead of this repo.
There are various Windows builds in the DT forum thread that range from mostly working to not working at all for different people.

**Comment by int-ua** [comment]
Can you (or someone else) explain what's the major difference and why isn't it pulled into upstream?

**Comment by lethosor** [comment]
splinterz hasn't been active for 6 months, and current 64-bit builds of DT are still unstable.

**Comment by wookietreiber** [comment]
I have been running with the [Hello71 version](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=dwarftherapist-git) for a while now on 64-bit Arch Linux and had no problems.

**Comment by anders-larsson** [comment]
I've been running Hello71's version as well. It's working quite well.

However once when I created a squad (using the Archer, default, armour set and Dwarf Therapist kept crashing. It only seemed to affect that armour set specifically because I could create a squad using a different set without any issues and I could even switch to Archer for the existing squid without any issues with Dwarf Therapist.

**Comment by lethosor** [comment]
I meant that current 64-bit builds of DT are unstable *for some people*. Sure, they can work fine for some people, but others are having issues such as
* DT crashing on startup, or refusing to start at all
* DT crashing when reading dwarves
* DT crashing when making certain edits to dwarves
* DT failing to make certain edits properly (maybe leading to DF crashing)

If it works for you, that's great for you, but it doesn't mean it's stable for everyone else. (Hello71's fork may be better on Linux than the various Windows builds are on Windows, though - I haven't tried it.)
